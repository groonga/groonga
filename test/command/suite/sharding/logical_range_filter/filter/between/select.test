#@on-error omit
plugin_register sharding
#@on-error default

table_create Logs_20150205 TABLE_NO_KEY
column_create Logs_20150205 timestamp COLUMN_SCALAR Time
column_create Logs_20150205 memo COLUMN_SCALAR ShortText

table_create Timestamp_20150205 TABLE_PAT_KEY Time
column_create Timestamp_20150205 index COLUMN_INDEX Logs_20150205 timestamp

load --table Logs_20150205
[
{
  "timestamp": "2015-02-05 13:49:00",
       "memo": "2015-02-05 13:49:00"
},
{
  "timestamp": "2015-02-05 13:50:00",
       "memo": "2015-02-05 13:50:00"
},
{
  "timestamp": "2015-02-05 13:51:00",
       "memo": "2015-02-05 13:51:00"
},
{
  "timestamp": "2015-02-05 13:52:00",
       "memo": "2015-02-05 13:52:00"
}
]

#@add-important-log-levels debug
log_level --level debug
#@collect-query-log true
logical_range_filter Logs timestamp \
  --limit -1 \
  --filter 'between(timestamp, "2015-02-05 13:50:00", "include", "2015-02-05 13:52:00", "exclude")'
#@collect-query-log false
log_level --level notice
#@remove-important-log-levels debug
