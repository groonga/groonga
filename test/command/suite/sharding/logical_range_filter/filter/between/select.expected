plugin_register sharding
[[0,0.0,0.0],true]
table_create Logs_20150205 TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Logs_20150205 timestamp COLUMN_SCALAR Time
[[0,0.0,0.0],true]
column_create Logs_20150205 memo COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Timestamp_20150205 TABLE_PAT_KEY Time
[[0,0.0,0.0],true]
column_create Timestamp_20150205 index COLUMN_INDEX Logs_20150205 timestamp
[[0,0.0,0.0],true]
load --table Logs_20150205
[
{
  "timestamp": "2015-02-05 13:49:00",
       "memo": "2015-02-05 13:49:00"
},
{
  "timestamp": "2015-02-05 13:50:00",
       "memo": "2015-02-05 13:50:00"
},
{
  "timestamp": "2015-02-05 13:51:00",
       "memo": "2015-02-05 13:51:00"
},
{
  "timestamp": "2015-02-05 13:52:00",
       "memo": "2015-02-05 13:52:00"
}
]
[[0,0.0,0.0],4]
log_level --level debug
[[0,0.0,0.0],true]
logical_range_filter Logs timestamp   --limit -1   --filter 'between(timestamp, "2015-02-05 13:50:00", "include", "2015-02-05 13:52:00", "exclude")'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        "memo",
        "ShortText"
      ],
      [
        "timestamp",
        "Time"
      ]
    ],
    [
      "2015-02-05 13:50:00",
      1423111800.0
    ],
    [
      "2015-02-05 13:51:00",
      1423111860.0
    ]
  ]
]
#|d| [logical_range_filter][select] <Logs_20150205>: limit is negative: <-1>
#>logical_range_filter --filter "between(timestamp, \"2015-02-05 13:50:00\", \"include\", \"2015-02-05 13:52:00\", \"exclude\")" --limit "-1" --logical_table "Logs" --shard_key "timestamp"
#:000000000000000 filter(2): between(Logs_20150205.timestamp, "2015-02-05 13:50:00", "include", "2015-02-05 13:52:00", "exclude")
#:000000000000000 sort(2)[Logs_20150205]: timestamp
#:000000000000000 output(2)
#<000000000000000 rc=0
log_level --level notice
[[0,0.0,0.0],true]
